### ISSUE TITLE
Error when trying to submit a job using MassStorageFile and {'sjid'}

### ISSUE THREAD

**Issue by jea-umd** [ganga-devs/ganga#120]
When trying to submit a new job in 6.1.14 which uses MassStorageFiles and the {'sjid'} flag, the following error is returned:

JobError: Error in MassStorageFile.outputfilenameformat field :  job is non-split, but {'sjid'} keyword found

This occurs even in the same Ganga session that was used to create the job.

**Comment by rob-c** [comment]
@jea-umd Firstly, are you submitting a job with a splitter and observing this?
Secondly, would you be able to provide a script to reproduce this?

Thanks,
Rob

**Comment by jea-umd** [comment]
Hi Rob,

Yes, I'm using a splitter. A simple ganga script which reproduces the 
error is attached.

cheers!
Jason

**Comment by milliams** [comment]
@jea-umd It seems that email attachments don't work with GitHub issues. Could you paste it into a comment box at https://github.com/ganga-devs/ganga/issues/120?

**Comment by jea-umd** [comment]
Here they are.

[gangaTmp.zip](https://github.com/ganga-devs/ganga/files/94653/gangaTmp.zip)

**Comment by mboubdir** [comment]
Hey everyone, 

Could you could fix this problem? I m also facing this since starting to use Ganga 6.1.14. Using other versions with the same submit script seems to work fine. Although, the upload to eos doesn't work anymore.

Cheers, 
Meriem

**Comment by rob-c** [comment]
Has this already been fixed? I know we have looked at a problem very similar to this in the past or is there a MWE which demonstrates what is broken?

**Comment by mboubdir** [comment]
We had a similar problem with Ganga 6.1.13 while resubmitting failed jobs/subjobs. We have been told that the problem will be fixed in 1.14 but now we get the error right after submitting a new job. I didn't find any solution yet.

**Comment by rob-c** [comment]
@mboubdir Can you provide 4 or 5 lines which reproduce this?

**Comment by mboubdir** [comment]
Ofc, here is a simplified script:


filepath =  '/LHCb/Collision15/Beam6500GeV-VeloClosed-MagDown/Real Data/Reco15a/Stripping23r1/90000000/BHADRONCOMPLETEEVENT.DST'

v = 'v38r0'
dv = DaVinci(version=v)
dv.optsfile=['options.py']

j = Job(application=dv)
j.backend = Dirac()
j.inputdata = data 

j.splitter = SplitByFiles(filesPerJob=100)
f = MassStorageFile('DVTuple.root')
f.outputfilenameformat = '/DVTuple/{sjid}_{fname}'
j.outputfiles +=[f]

j.submit()

**Comment by rob-c** [comment]
Does this occur without using the Dirac backend or without DaVinci do you know?

**Comment by mboubdir** [comment]
I Don't know and can't try it now.

**Comment by jea-umd** [comment]
I just tried mboubdir's recipe replacing DaVinci with Moore v23r7p5 and 
replacing Dirac with Local ( on lxplus ), and I get the identical error.

Jason

**Comment by mboubdir** [comment]
@jea-umd Thank you! 

Meriem

**Comment by egede** [comment]
Here is a a simpler example to reproduce the error.

```
j=Job()
j.application=Executable(exe='touch')
j.splitter=ArgSplitter(args=[['abc.txt'], ['def.txt']])
j.outputfiles=[MassStorageFile(outputfilenameformat = '/test/{sjid}-{fname}', namePattern = '*.txt')]
j.submit()
```

**Comment by rob-c** [comment]
@egede Thanks for that example, maybe we can get this fixed for 6.1.19 I suspect I may know where this is causing problems but I'd have to do some digging.
